### PR TITLE
Fix `gluonts.dataset.split` for multivariate case

### PR DESCRIPTION
**Pull request by lostella**
*Description of changes:* Fixes to `gluonts.dataset.split` in the multivariate target case, and added test cases

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: src/gluonts/dataset/rolling_dataset.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def truncate_features(timeseries: dict, max_len: int) -> dict:
     ):
         if key not in timeseries:
             continue
-        timeseries[key] = [feature[:max_len] for feature in timeseries[key]]
+        timeseries[key] = timeseries[key][:, :max_len]
```
Comment (jaheba) [inline src/gluonts/dataset/rolling_dataset.py:114]: How is this related to the split?

File: src/gluonts/dataset/split.py [hunk 2]
```diff
@@ -137,7 +137,7 @@ def slice_data_entry(
 ) -> DataEntry:
     slice_ = to_positive_slice(
         to_integer_slice(slice_, entry[FieldName.START]),
-        len(entry[FieldName.TARGET]),
+        entry[FieldName.TARGET].shape[-1],
```
Comment (jaheba) [inline src/gluonts/dataset/split.py:140]: I think the named axis approach could be handy here.

File: src/gluonts/dataset/split.py [hunk 3]
```diff
@@ -152,10 +152,16 @@ def slice_data_entry(
     if slice_.start is not None:
         offset = slice_.start
         if offset < 0:
-            offset += entry["target"].shape[0]
+            offset += entry["target"].shape[-1]
         sliced_entry[FieldName.START] += offset
 
-    sliced_entry[FieldName.TARGET] = sliced_entry[FieldName.TARGET][slice_]
+    # TODO fix
```
Comment (jaheba) [inline src/gluonts/dataset/split.py:158]: What would the fix be?

**Review by jaheba — APPROVED** [review]
I think this is another proof that our schema is too fixed.

File: src/gluonts/dataset/rolling_dataset.py (hunk 1 above)
Comment (lostella) [reply src/gluonts/dataset/rolling_dataset.py:114]: I think it just ended up there. In any way, it's a fix :-)

File: src/gluonts/dataset/split.py (hunk 3 above)
Comment (lostella) [reply src/gluonts/dataset/split.py:158]: I was thinking of getting rid of the branching in favor of constructing a better `slice_` that would work in all cases. But it's OK for now
